Mathlib/Topology/Constructions.lean

```diff
@@ -1075,6 +1076,115 @@ lemma IsClosedEmbedding.sumElim {f : X → Z} {g : Y → Z}
   rw [IsClosedEmbedding.isClosedEmbedding_iff_continuous_injective_isClosedMap] at hf hg ⊢
   exact ⟨hf.1.sumElim hg.1, h, hf.2.2.sumElim hg.2.2⟩
 
+section IsInducing
+
+variable {f : X → Z} {g : Y → Z}
+
+/-- If `Sum.elim f g` is an inducing map, then so is `f`. -/
+lemma Topology.IsInducing.sumElim_left (h : IsInducing (Sum.elim f g)) : IsInducing f :=
+  elim_comp_inl f g ▸ h.comp IsEmbedding.inl.isInducing
+
+/-- If `Sum.elim f g` is an inducing map, then so is `g`. -/
+lemma Topology.IsInducing.sumElim_right (h : IsInducing (Sum.elim f g)) : IsInducing g :=
+  elim_comp_inr f g ▸ h.comp IsEmbedding.inr.isInducing
+
+/-- If `f` and `g` are inducing maps whose ranges are separated, then `Sum.elim f g` is inducing. -/
+theorem Topology.IsInducing.sumElim (hf : IsInducing f) (hg : IsInducing g)
+    (hFg : Disjoint (closure (range f)) (range g)) (hfG : Disjoint (range f) (closure (range g))) :
+    IsInducing (Sum.elim f g) := by
+  rw [isInducing_iff_nhds]
+  intro x
+  apply le_antisymm ((hf.continuous.sumElim hg.continuous).tendsto x).le_comap
+  intro s hs
+  rw [mem_comap_iff_compl, ← image_preimage_inl_union_image_preimage_inr sᶜ]
```

Comment:
@plp127 This proof is a bit long, so having a comment outlining the overall strategy would help.
Do you have a moment to sketch the strategy of proving this, say in a comment? This would help a lot (also for deciding if this proof can be golfed more: I don't understand the high-level argument *yet*.)